datafusion/spark/src/function/datetime/date_add.rs

```diff
@@ -88,10 +90,12 @@ impl ScalarUDFImpl for SparkDateAdd {
 
 fn spark_date_add(args: &[ArrayRef]) -> Result<ArrayRef> {
     let [date_arg, days_arg] = args else {
-        return internal_err!(
-            "Spark `date_add` function requires 2 arguments, got {}",
-            args.len()
+        assert_eq_or_internal_err!(
+            args.len(),
+            2,
+            "Spark `date_add` function requires 2 arguments"
```

Comment:
```suggestion
            "Spark `date_add` function requires 2 arguments, got {}"
```
This is useful while debugging. Please keep it!